crates/polars-ops/src/chunked_array/list/namespace.rs

```diff
@@ -571,6 +572,96 @@ pub trait ListNameSpaceImpl: AsList {
         list_ca.apply_amortized(|s| s.as_ref().drop_nulls())
     }
 
+    #[cfg(feature = "list_pad")]
+    fn lst_pad_start(&self, fill_value: &Column, length: &Column) -> PolarsResult<ListChunked> {
+        let ca = self.as_list();
+        let inner_dtype = ca.inner_dtype();
+        let fill_dtype = fill_value.dtype();
+        let super_type = try_get_supertype(inner_dtype, fill_dtype)?;
+
+        let dtype = &DataType::List(Box::new(super_type.clone()));
+        let ca = ca.cast(dtype)?;
+        let ca = ca.list().unwrap();
+
+        let length = if length.len() == 1 {
+            &length.new_from_index(0, ca.len())
+        } else {
+            length
+        };
+        let length = length.strict_cast(&IDX_DTYPE)?;
+        let mut length = length.idx()?.into_iter();
+
+        let fill_value = if fill_value.len() == 1 {
+            &fill_value.new_from_index(0, ca.len())
+        } else {
+            fill_value
+        };
+        let fill_value = fill_value.cast(&super_type)?;
+
+        let fill_value_series = fill_value.as_materialized_series();
+        let mut builder = get_list_builder(
+            &super_type,
+            ca.get_values_size() + 1000,
+            ca.len(),
+            ca.name().clone(),
+        );
+
+        for (idx, opt_s) in ca.iter().enumerate() {
+            let target_length = match length.next().unwrap() {
+                Some(len) => len as usize,
+                None => {
+                    return Err(PolarsError::InvalidOperation(
+                        "negative length not supported".into(),
+                    ));
+                },
+            };
+
+            match opt_s {
+                Some(s) => {
+                    let s_series = Series::from_arrow(PlSmallStr::EMPTY, s)?;
+
+                    match target_length.cmp(&s_series.len()) {
+                        Ordering::Equal | Ordering::Less => {
+                            let s_casted = s_series.cast(&super_type).unwrap();
+                            builder.append_series(&s_casted).unwrap();
+                        },
+                        Ordering::Greater => {
+                            let pad_count = target_length - s_series.len();
+
+                            let fill_val = fill_value_series.get(idx).unwrap();
+                            let fill_series = Series::from_any_values(
+                                PlSmallStr::EMPTY,
+                                &vec![fill_val; pad_count],
```

Comment:
This should be less allocating, like going through vec and allocating for every element is not good enough for performance.